python/ray/serve/storage/kv_store.py

```diff
@@ -174,32 +174,28 @@ class RayS3KVStore(KVStoreBase):
     caller must handle serialization.
     """
 
-    def __init__(
-            self,
-            namepsace: str,
-            bucket="",
-            prefix="",
-            region_name="us-west-2",
-            aws_access_key_id=None,
-            aws_secret_access_key=None,
-            aws_session_token=None,
-    ):
+    def __init__(self,
+                 namepsace: str,
+                 bucket="",
+                 prefix="",
+                 region_name="us-west-2"):
         self._namespace = namepsace
         self._bucket = bucket
-        self._prefix = prefix + "/" if prefix else ""
+        self._prefix = prefix
         if not boto3:
             raise ImportError(
                 "You tried to use S3KVstore client without boto3 installed."
                 "Please run `pip install boto3`")
         self._s3 = boto3.client(
             "s3",
             region_name=region_name,
-            aws_access_key_id=aws_access_key_id,
-            aws_secret_access_key=aws_secret_access_key,
-            aws_session_token=aws_session_token)
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID", None),
```

Comment:
i think previously we're not really getting these tokens in make_kv_store, so adding it at client level with actual s3 integration.